src/prompts/README.md

```diff
@@ -0,0 +1,294 @@
+const builder = new AutoblocksPromptBuilder(PromptTrackingId.FEATURE_A);
+
+// Use the builder to build prompt(s) programatically
+// using any combination of templates you want
+const messages = [
+  {
+    role: 'system',
+    content: builder.build('feature-a/system', {
+      // Replacement value for the {{ languageRequirement }} placeholder
+      // is the result of another rendered template
+      languageRequirement: builder.build('common/language', {
+        // Replacement value for the {{ language }} placeholder is a constant
+        language: 'Spanish',
+      }),
+    }),
+  },
+  {
+    role: 'user',
+    content: builder.build('feature-a/user', {
+      // Replacement value for the {{ name }} placeholder is a constant
+      name: 'Alice',
+    }),
+  },
+];
+
+const response = await openai.chat.completions.create({
+  messages,
+  model: 'gpt-3.5-turbo',
+});
+
+// Record response + template usage to Autoblocks
+await tracer.sendEvent('ai.response', {
+  properties: { response },
+  promptTracking: builder.usage(),
+});
+```
+
+If you make a change to **any** of the below templates, it will result in a new version for `PromptTrackingId.FEATURE_A`:
+
+- `common/language`
+- `feature-a/system`
+- `feature-a/user`
+
+Additionally, if you introduce a new template, it would also result in a new version:
+
+```ts
+const builder = new AutoblocksPromptBuilder(PromptTrackingId.FEATURE_A);
+
+const messages = [
+  {
+    role: 'system',
+    content: builder.build('feature-a/system', {
+      languageRequirement: builder.build('common/language', {
+        language: 'Spanish',
+      }),
+      // New template, so we'll see a new version for FEATURE_A
+      // when we deploy this change.
+      toneRequirement: builder.build('common/tone', {
+        tone: 'friendly',
+      }),
+    }),
+  },
+  {
+    role: 'user',
+    content: builder.build('feature-a/user', {
+      name: 'Alice',
+    }),
+  },
+];
+```
+
+# Getting Started
+
+## Install
+
+```bash
+npm install @autoblocks/client
+```
+
+```bash
+yarn add @autoblocks/client
+```
+
+```bash
+pnpm add @autoblocks/client
+```
+
+## Configure your templates directory
+
+Tell us where your templates directory is:
+
+`package.json`:
+
+```json
+"autoblocks": {
+  "templatesDirectory": "autoblocks-templates"
+}
+```
+
+## Create templates
+
+Add templates to your templates directory.
+
+```
+package.json
+autoblocks-templates/
+  common/
+    language
+    tone
+  feature-a/
+    system
+    user
+  feature-b/
+    system
+    user
+```
+
+These are just plain text files with placeholders. Placeholders should be surrounded by double curly braces:
+
+```txt
+This is a template with a {{ placeholder }}.
+```
+
+Placeholders can have arbitrary whitespace, so this is also valid:
+
+```txt
+My name is {{name}}.
+My age is {{      age       }}
+```
+
+> **_HINT:_** Placeholders aren't always replaced with constants. They can also be replaced with the value of another rendered template.
```

Comment:
can we add an example of this? Like language or point to that.

Reply:
Ya I can update when I move to the docs